src/app/components/card-list/card-list.component.ts

```diff
@@ -0,0 +1,154 @@
+import { SearchBarEventArgs } from '../nav-bar/nav-bar.component';
+import { FavoriteEntityService } from 'src/app/services/favorite-pokemons/favorite-entity.service';
+
+@Component({
+  selector: 'app-card-list',
+  templateUrl: './card-list.component.html',
+  styleUrls: ['./card-list.component.scss'],
+})
+export class CardListComponent implements OnInit {
+  pokemonsPage$: Observable<PokemonsPage[]>;
+  isComparing: boolean = false;
+  pokemonBeforeComparing: Pokemon;
+  pokemonApiOffset: string = '20';
+  queryParams: string = '';
+
+  constructor(
+    private pokemonsEntityService: PokemonEntityService,
+    private dialog: MatDialog,
+    private fetchedPokemonsEntityService: FetchedPokemonsEntityService,
+    private favoriteEntityService: FavoriteEntityService,
+    private loader: NgxUiLoaderService
+  ) {}
+
+  ngOnInit(): void {
+    this.reload();
+
+    this.pokemonsEntityService.entities$.forEach((item) => {
+      this.pokemonApiOffset = item.length + '';
+    });
+  }
+
+  reload() {
+    if (this.queryParams !== '' && this.queryParams !== undefined) {
+      this.pokemonsPage$ = this.pokemonsEntityService.entities$.pipe(
+        map((pokemons) =>
+          pokemons.filter((item) => item.name.includes(this.queryParams))
+        )
+      );
+    } else {
+      this.pokemonsPage$ = this.pokemonsEntityService.entities$.pipe(
+        map((pokemons) => pokemons)
+      );
+    }
+  }
+
+  filterData(data: SearchBarEventArgs) {
+    this.queryParams = data.newValue;
+    this.reload();
+  }
+
+  addToFavorites(pokemon: PokemonsPage) {
+    let favoritesLength = undefined;
+    let removingFavorite = false;
+
+    this.favoriteEntityService.entities$.subscribe((entities) => {
+      favoritesLength = entities.length;
+
+      if (entities.find((item) => item.name == pokemon.name)) {
+        removingFavorite = true;
+      }
+    });
+    if (removingFavorite) {
+      const newPokemon = { ...pokemon, isFavorite: false };
+      this.pokemonsEntityService.updateOneInCache(newPokemon);
+      this.favoriteEntityService.removeOneFromCache(pokemon);
+    } else {
+      if (favoritesLength !== undefined && favoritesLength >= 5) {
+        console.log(`ya no agregamos mas`);
+      } else {
+        const newPokemon = { ...pokemon, isFavorite: true };
+        this.pokemonsEntityService.updateOneInCache(newPokemon);
+        this.favoriteEntityService.addOneToCache(pokemon);
+      }
+    }
+  }
+
+  loadMorePokemons() {
+    this.pokemonsEntityService.getWithQuery({
+      limit: '20',
+      id: this.pokemonApiOffset,
+    });
+    let nextOffsetInt = parseInt(this.pokemonApiOffset) + 20;
+    this.pokemonApiOffset = nextOffsetInt + '';
+  }
+
+  async openPokemonDialog(pokemon) {
+    const dialogConfig = defaultDialogConfig();
+
+    let pokemonData: Pokemon = undefined;
+    this.fetchedPokemonsEntityService.entities$.subscribe((entity) => {
```

Comment:
this is an observable and the data is async, when using a async make sure you await for the result of the calls.
I don't recommend using async as it limits observables to be promises